public/login.js

```diff
@@ -29,14 +29,7 @@ export function login() {
   const googlelogin = document.getElementById("googlelogin");
 
   // login
-  login.addEventListener("click", (e) => {
-    const auth = firebase.auth();
-    const promise = auth.signInWithEmailAndPassword(
-      email.value,
-      password.value
-    );
-    promise.catch((e) => console.log(e.message));
-  });
+  // login.addEventListener("click");

```

Comment:
At this point, `login` is undefined so this was crashing this component. It happens for all the elements that you are trying to get.
You can follow an alternative approach and define the onClick event directly on the element itself (you can check that I moved the function you had here to line 105.
This way you are sure that the element is defined.